### PR TITLE
MNT: update repository

### PR DESCRIPTION
**Pull request by kmuehlbauer**
- remove Python 2 compatibility
- switch to `h5netcdf` org
- switch from travis to github actions
- use black style, isort and flake8 testing
- test both h5py 2 and 3

### CODE REVIEW

**Comment by kmuehlbauer** [comment]
@shoyer This is the first shot which works quite nice in [my fork](https://github.com/kmuehlbauer/h5netcdf/actions). Please have a look over there before merging. I've taken the chance to also add checks for lint and style (needed to reformat files). Hope this fits your requirements.

After this is merged I'll update my other PR to get h5py>=3 resolved.

File: h5netcdf/legacyapi.py [hunk 1]
```diff
@@ -1,7 +1,8 @@
 import h5py
 
 from . import core
-from .compat import unicode
+
+# from .compat import unicode
```
Comment (shoyer) [inline h5netcdf/legacyapi.py:5]: Delete?

File: h5netcdf/legacyapi.py [hunk 2]
```diff
@@ -31,32 +33,34 @@ def __setattr__(self, name, value):
...
+            "zlib": self._h5ds.compression == "gzip",
+        }
 
     @property
     def dtype(self):
         dt = self._h5ds.dtype
-        if h5py.check_dtype(vlen=dt) is unicode:
-            return str
+        # if h5py.check_dtype(vlen=dt) is unicode:
+        #    return str
```
Comment (shoyer) [inline h5netcdf/legacyapi.py:58]: I think we still want this for compatibility with netCDF4-python?

File: h5netcdf/tests/test_h5netcdf.py [hunk 3]
```diff
@@ -101,223 +111,234 @@ def is_h5py_char_working(tmp_netcdf, name):
...
     assert v.ndim == 0
     assert v.dimensions == ()
     assert v.ncattrs() == []
 
-    v = ds.variables['var_len_str']
-    assert v.dtype == str
-    assert v[0] == u'foo'
+    v = ds.variables["var_len_str"]
+    print("dtype:", v.dtype)
+    # assert v.dtype == 'O'
```
Comment (shoyer) [inline h5netcdf/tests/test_h5netcdf.py:284]: This is the test for the commented out inline setting "str" as a dtype

**Review by shoyer — COMMENTED** [review]
Thanks for cleanup PR!

File: h5netcdf/legacyapi.py (hunk 1 above)
Comment (kmuehlbauer) [reply h5netcdf/legacyapi.py]: Sure!

File: h5netcdf/legacyapi.py (hunk 2 above)
Comment (kmuehlbauer) [reply h5netcdf/legacyapi.py]: Yes, forgot to fix this one...

File: h5netcdf/tests/test_h5netcdf.py (hunk 3 above)
Comment (kmuehlbauer) [reply h5netcdf/tests/test_h5netcdf.py]: And forgot this one too...

**Comment by kmuehlbauer** [comment]
@shoyer I've corrected the things per your suggestion. Removed the h5py3 fix. Added h5py3 tests to the matrix (to see things breaking).

I think this now ready for review/merge and we can move on to #82.

**Review by shoyer — APPROVED** [review]

**Comment by shoyer** [comment]
thanks!

**Pull request merged**